deploy/crds/k8gb.absa.oss_gslbs_crd.yaml

```diff
@@ -65,6 +65,8 @@ spec:
                       under a specified host to the related backend services. Incoming
                       requests are first evaluated for a host match, then routed to
                       the backend associated with the matching IngressRuleValue.
+                    required:
+                    - http
```

Comment:
Hi, love that.

The code looks like this:
```go
	for _, gslb := range gslbList.Items {
				for _, rule := range gslb.Spec.Ingress.Rules {
					for _, path := range rule.HTTP.Paths {
						if path.Backend.ServiceName == a.Meta.GetName() {
							gslbName = gslb.Name
						}
					}
				}
```
My question: The code will work if `HTTP.Paths`  is empty (for cycle will be skipped), but isn't `Path` mandatory as well?  @Tim, @k0da

Reply:
@kuritka Thanks, definetely we need that too.

```shell
09:04:27 ❯ kubectl apply -f terratest/examples/invalid.yaml
The Gslb "invalid-gslb" is invalid: spec.ingress.rules.http.paths: Required value
```

Reply:
As I mentioned earlier, nor `http` neither `paths` are not enforced by Ingress API, and k8gb should stick to that, if it wants to resemble Ingress-like experience for end-users.
k8gb supports indirect `gslb` object creation Ingress object with k8gb-specific annotations, and you won't get any errors if `http` or `path` sections are missing in Ingress manifest.
In that sense it is better to be prepared for missing spec sessions in the code instead of imposing restrictions in CRD.

Reply:
But it is required for GSLB IIUC.

Reply:
In this case gslb creation can be skipped with log message, like in case when no glsb manifests are found or when `glsb.Name` is empty

Reply:
Ok, we can go this way instead. @ytsarev @kuritka @donovanmuller  what do you think about that?

Reply:
Direct CRD modification is an elegant way to validate on the other hand, there was a decision to follow standard ingress definition.
When I hastily approved, I haven't realized the only input point in our architecture is depresolver, which has two parts - config and spec. Unfortunately, I'm afraid we can't break the validation into more places. It would be difficult to find out where the various inputs are being validated / tested.

Reply:
Actually, I was not correct about `spec.ingress.rules.http.paths`. It mandatory field by Ingress spec:
```yaml
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  annotations:
    k8gb.io/strategy: roundRobin
  name: invalid-ingress
spec:
  rules:
  - host: invalid-ingress.example.com
    http:
      paths:
```
```sh
k apply -f invalid-ingress.yaml --validate=true
error: error validating "invalid-ingress.yaml": error validating data: ValidationError(Ingress.spec.rules[0].http): missing required field "paths" in io.k8s.api.extensions.v1beta1.HTTPIngressRuleValue; if you choose to ignore these errors, turn validation off with --validate=false
```
So it's OK to impose this rule on CRD level:
```yaml
required:
- paths
```

but `spec.ingress.rules.http` is still not required:
```yaml
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  annotations:
    k8gb.io/strategy: roundRobin
  name: invalid-ingress
spec:
  rules:
  - host: invalid-ingress.example.com
    http:
```
```sh
k apply -f invalid-ingress.yaml
ingress.extensions/invalid-ingress created
```